lms/services/assignment.py

```diff
@@ -238,6 +241,11 @@ def get_assignments(self, h_userid: str | None = None) -> Select[tuple[Assignmen
                 .where(User.h_userid == h_userid)
             )
 
+        if course_id:
+            assignments_query = assignments_query.join(AssignmentGrouping).where(
+                AssignmentGrouping.grouping_id == course_id
+            )
```

Comment:
How does this work with assignments that use groups and/or sections?

Reply:
Assignments that use groupings will have rows in AssignmentGrouping for the course and each of the segments (either sections or groups).

But IMO while we are using the same table to represent both courses and segments, at the API level we should expose two different filters when we eventually expose segments in the dashboard.